### PR TITLE
feat: save action validates inputs

### PR DESCRIPTION
**Pull request by cgdibble**
### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues

DTNOR-546

These are part of the progressive implementation of the Save action. These changes enable the action to:
- be initialized
- accept config from merchant
- validate that config
- return skeleton of the action (implemented in tickets to come)

Additionally, this PR adds unit testing and the tools to use it to this repository.

### Reproduction Steps (if applicable)
n/a

### Screenshots (if applicable)
n/a

### Dependent Changes (if applicable)
n/a

### Groups who should review (if applicable)

❤️  Thank you!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "release": "./scripts/publish.sh",
     "start": "npm run webpack -- --progress --watch",
     "test": "npm run jest-ssr && npm run karma && npm run jest-screenshot",
```
Comment (wsbrunson) [inline package.json:29]:
```suggestion
    "test": "npm run test:unit && npm run jest-ssr && npm run karma && npm run jest-screenshot",
```

**Review by wsbrunson — APPROVED** [review]

File: src/actions/save/index.js [hunk 2]
```diff
@@ -1,17 +1,46 @@
...
+  createVaultSetupToken: () => Promise<string>,
+  onApprove: ({| vaultSetupToken: string |}) => void,
 |};
 
-export type SaveAction = (SaveActionHandlers) => ({|
+export type SaveAction = (SaveActionConfig) => ({|
   type: "save",
-  ...SaveActionHandlers,
+  /* TODO: 
+      - We need to define how paymentSourceDetails is typed here
```
Comment (cgdibble) [inline src/actions/save/index.js:12]: A subsequent ticket will handle the definition of paymentSourceDetails. We'll type out the current expectation based on card fields data then, in the future, provide others as options.

File: src/actions/save/index.js [hunk 3]
```diff
@@ -1,17 +1,46 @@
...
+      throw new ValidationError(`Save action is missing the required '${inputProp}' callback`)  
+    }
+  }
+}
+
+/**
+ * Creating a Save action allows us to validate initial inputs from the merchant, and then return the resulting object. 
+ */
+export const createSaveAction: SaveAction = (config: SaveActionConfig) => {
+  const { createVaultSetupToken } = config;
```
Comment (thebastedo) [inline src/actions/save/index.js:38]: Would we want to validate prior to destructuring anything off of the config? Maybe it's a bit early to get that picky about the code?

Comment (cgdibble) [reply src/actions/save/index.js:38]: I need to push some changes, but yeah this is just left from when I did it here initially and destructured the props. I forgot to delete this line when I moved validation to a function, so yeah it'll be removed here shortly

**Review by thebastedo — APPROVED** [review]

**Pull request merged**